### PR TITLE
[Feat/#37]:목록 조회 추가 수정

### PR DESCRIPTION
**Pull request by juhee067**
## 📌 작업 내용

- header 고정
- 목록 조회 grid 수정
- 늦게 뜨는 이미지 임시 스피너 추가
- params camelCase로 수정

<br/>

## 📌 구현 결과 (선택)
화면-기록-2024-08-08-22 39 25
<br/>

## 📌 기타 사항

TODO
- [x] header 고정

고민
- 너비에 따라 딱맞게 떨어지지않는 카드 어떤식으로 가지고와야할지
- 너무 큰 이미지 탓에 이미지 늦게 뜨는 문제 어떻게 해결해야할지, 이미지를 webp으로 받으면 좋을지

<br/>

### CODE REVIEW

**Comment by JIMIN1020** [comment]
<h2>Coverage Report</h2>
<table> <thead> <tr> <th align="center">Status</th> <th align="left">Category</th> <th align="right">Percentage</th> <th align="right">Covered / Total</th> </tr> </thead> <tbody> <tr> <td align="center">🔵</td> <td align="left">Lines</td> <td align="right">0%</td> <td align="right">0 / 4094</td> </tr> <tr> <td align="center">🔵</td> <td align="left">Statements</td> <td align="right">0%</td> <td align="right">0 / 4094</td> </tr> <tr> <td align="center">🔵</td> <td align="left">Functions</td> <td align="right">3.31%</td> <td align="right">5 / 151</td> </tr> <tr> <td align="center">🔵</td> <td align="left">Branches</td> <td align="right">3.31%</td> <td align="right">5 / 151</td> </tr> </tbody> </table><details><summary>File Coverage</summary><table> <thead> <tr> <th align="left">File</th> <th align="right">Stmts</th> <th align="right">% Branch</th> <th align="right">% Funcs</th> <th align="right">% Lines</th> <th align="left">Uncovered Lines</th> </tr> </thead> <tbody> <tr> <td align="left" colspan="6"><b>Changed Files</b></td> </tr> <tr> <td align="left"><a href="https://github.com/Licruit/Licruit-frontend/blob/e49b752801f61e33e214508d8424422d63cd81f9/src/features/Catalog/components/Filter.tsx">src/features/Catalog/components/Filter.tsx</a></td> <td align="right">0%</td> <td align="right">0%</td> <td align="right">0%</td> <td align="right">0%</td> <td align="left"><a href="https://github.com/Licruit/Licruit-frontend/blob/e49b752801f61e33e214508d8424422d63cd81f9/src/features/Catalog/components/Filter.tsx#L1-L5" class="text-red">1-5</a>, <a href="https://github.com/Licruit/Licruit-frontend/blob/e49b752801f61e33e214508d8424422d63cd81f9/src/features/Catalog/components/Filter.tsx#L7-L18" class="text-red">7-18</a>, <a href="https://github.com/Licruit/Licruit-frontend/blob/e49b752801f61e33e214508d8424422d63cd81f9/src/features/Catalog/components/Filter.tsx#L20-L23" class="text-red">20-23</a>, <a href="https://github.com/Licruit/Licruit-frontend/blob/e49b752801f61e33e214508d8424422d63cd81f9/src/features/Catalog/components/Filter.tsx#L25-L34" class="text-red">25-34</a>, <a href="https://github.com/Licruit/Licruit-frontend/blob/e49b752801f61e33e214508d8424422d63cd81f9/src/features/Catalog/components/Filter.tsx#L36-L56" class="text-red">36-56</a>, <a href="https://github.com/Licruit/Licruit-frontend/blob/e49b752801f61e33e214508d8424422d63cd81f9/src/features/Catalog/components/Filter.tsx#L58-L59" class="text-red">58-59</a>, <a href="https://github.com/Licruit/Licruit-frontend/blob/e49b752801f61e33e214508d8424422d63cd81f9/src/features/Catalog/components/Filter.tsx#L61-L64" class="text-red">61-64</a>, <a href="https://github.com/Licruit/Licruit-frontend/blob/e49b752801f61e33e214508d8424422d63cd81f9/src/features/Catalog/components/Filter.tsx#L66-L68" class="text-red">66-68</a>, <a href="https://github.com/Licruit/Licruit-frontend/blob/e49b752801f61e33e214508d8424422d63cd81f9/src/features/Catalog/components/Filter.tsx#L70" class="text-red">70</a>, <a href="https://github.com/Licruit/Licruit-frontend/blob/e49b752801f61e33e214508d8424422d63cd81f9/src/features/Catalog/components/Filter.tsx#L72" class="text-red">72</a>, <a href="https://github.com/Licruit/Licruit-frontend/blob/e49b752801f61e33e214508d8424422d63cd81f9/src/features/Catalog/components/Filter.tsx#L74" class="text-red">74</a>, <a href="https://github.com/Licruit/Licruit-frontend/blob/e49b752801f61e33e214508d8424422d63cd81f9/src/features/Catalog/components/Filter.tsx#L76" class="text-red">76</a>, <a href="https://github.com/Licruit/Licruit-frontend/blob/e49b752801f61e33e214508d8424422d63cd81f9/src/features/Catalog/components/Filter.tsx#L78-L80" class="text-red">78-80</a>, <a href="https://github.com/Licruit/Licruit-frontend/blob/e49b752801f61e33e214508d8424422d63cd81f9/src/features/Catalog/components/Filter.tsx#L83-L84" class="text-red">83-84</a>, <a href="https://github.com/Licruit/Licruit-frontend/blob/e49b752801f61e33e214508d8424422d63cd81f9/src/features/Catalog/components/Filter.tsx#L87" class="text-red">87</a>, <a href="https://github.com/Licruit/Licruit-frontend/blob/e49b752801f61e33e214508d8424422d63cd81f9/src/features/Catalog/components/Filter.tsx#L91-L92" class="text-red">91-92</a>, <a href="https://github.com/Licruit/Licruit-frontend/blob/e49b752801f61e33e214508d8424422d63cd81f9/src/features/Catalog/components/Filter.tsx#L96-L97" class="text-red">96-97</a>, <a href="https://github.com/Licruit/Licruit-frontend/blob/e49b752801f61e33e214508d8424422d63cd81f9/src/features/Catalog/components/Filter.tsx#L100" class="text-red">100</a>, <a href="https://github.com/Licruit/Licruit-frontend/blob/e49b752801f61e33e214508d8424422d63cd81f9/src/features/Catalog/components/Filter.tsx#L102-L103" class="text-red">102-103</a></td> </tr> <tr> <td align="left"><a href="https://github.com/Licruit/Licruit-frontend/blob/e49b752801f61e33e214508d8424422d63cd81f9/src/features/Catalog/components/Header/CatalogHeader.tsx">src/features/Catalog/components/Header/CatalogHeader.tsx</a></td> <td align="right">0%</td> <td align="right">0%</td> <td align="right">0%</td> <td align="right">0%</td> <td align="left"><a href="https://github.com/Licruit/Licruit-frontend/blob/e49b752801f61e33e214508d8424422d63cd81f9/src/features/Catalog/components/Header/CatalogHeader.tsx#L1" class="text-red">1</a>, <a href="https://github.com/Licruit/Licruit-frontend/blob/e49b752801f61e33e214508d8424422d63cd81f9/src/features/Catalog/components/Header/CatalogHeader.tsx#L3-L4" class="text-red">3-4</a>, <a href="https://github.com/Licruit/Licruit-frontend/blob/e49b752801f61e33e214508d8424422d63cd81f9/src/features/Catalog/components/Header/CatalogHeader.tsx#L6-L14" class="text-red">6-14</a>, <a href="https://github.com/Licruit/Licruit-frontend/blob/e49b752801f61e33e214508d8424422d63cd81f9/src/features/Catalog/components/Header/CatalogHeader.tsx#L16" class="text-red">16</a>, <a href="https://github.com/Licruit/Licruit-frontend/blob/e49b752801f61e33e214508d8424422d63cd81f9/src/features/Catalog/components/Header/CatalogHeader.tsx#L18" class="text-red">18</a>, <a href="https://github.com/Licruit/Licruit-frontend/blob/e49b752801f61e33e214508d8424422d63cd81f9/src/features/Catalog/components/Header/CatalogHeader.tsx#L20" class="text-red">20</a>, <a href="https://github.com/Licruit/Licruit-frontend/blob/e49b752801f61e33e214508d8424422d63cd81f9/src/features/Catalog/components/Header/CatalogHeader.tsx#L23" class="text-red">23</a>, <a href="https://github.com/Licruit/Licruit-frontend/blob/e49b752801f61e33e214508d8424422d63cd81f9/src/features/Catalog/components/Header/CatalogHeader.tsx#L40-L42" class="text-red">40-42</a></td> </tr> <tr> <td align="left"><a href="https://github.com/Licruit/Licruit-frontend/blob/e49b752801f61e33e214508d8424422d63cd81f9/src/features/Catalog/components/Main/ProductCard.tsx">src/features/Catalog/components/Main/ProductCard.tsx</a></td> <td align="right">0%</td> <td align="right">0%</td> <td align="right">0%</td> <td align="right">0%</td> <td align="left"><a href="https://github.com/Licruit/Licruit-frontend/blob/e49b752801f61e33e214508d8424422d63cd81f9/src/features/Catalog/components/Main/ProductCard.tsx#L1-L4" class="text-red">1-4</a>, <a href="https://github.com/Licruit/Licruit-frontend/blob/e49b752801f61e33e214508d8424422d63cd81f9/src/features/Catalog/components/Main/ProductCard.tsx#L18-L19" class="text-red">18-19</a>, <a href="https://github.com/Licruit/Licruit-frontend/blob/e49b752801f61e33e214508d8424422d63cd81f9/src/features/Catalog/components/Main/ProductCard.tsx#L21-L25" class="text-red">21-25</a>, <a href="https://github.com/Licruit/Licruit-frontend/blob/e49b752801f61e33e214508d8424422d63cd81f9/src/features/Catalog/components/Main/ProductCard.tsx#L27-L32" class="text-red">27-32</a>, <a href="https://github.com/Licruit/Licruit-frontend/blob/e49b752801f61e33e214508d8424422d63cd81f9/src/features/Catalog/components/Main/ProductCard.tsx#L34-L36" class="text-red">34-36</a>, <a href="https://github.com/Licruit/Licruit-frontend/blob/e49b752801f61e33e214508d8424422d63cd81f9/src/features/Catalog/components/Main/ProductCard.tsx#L38" class="text-red">38</a>, <a href="https://github.com/Licruit/Licruit-frontend/blob/e49b752801f61e33e214508d8424422d63cd81f9/src/features/Catalog/components/Main/ProductCard.tsx#L40-L47" class="text-red">40-47</a>, <a href="https://github.com/Licruit/Licruit-frontend/blob/e49b752801f61e33e214508d8424422d63cd81f9/src/features/Catalog/components/Main/ProductCard.tsx#L49" class="text-red">49</a>, <a href="https://github.com/Licruit/Licruit-frontend/blob/e49b752801f61e33e214508d8424422d63cd81f9/src/features/Catalog/components/Main/ProductCard.tsx#L51-L53" class="text-red">51-53</a>, <a href="https://github.com/Licruit/Licruit-frontend/blob/e49b752801f61e33e214508d8424422d63cd81f9/src/features/Catalog/components/Main/ProductCard.tsx#L56" class="text-red">56</a>, <a href="https://github.com/Licruit/Licruit-frontend/blob/e49b752801f61e33e214508d8424422d63cd81f9/src/features/Catalog/components/Main/ProductCard.tsx#L63" class="text-red">63</a>, <a href="https://github.com/Licruit/Licruit-frontend/blob/e49b752801f61e33e214508d8424422d63cd81f9/src/features/Catalog/components/Main/ProductCard.tsx#L77" class="text-red">77</a>, <a href="https://github.com/Licruit/Licruit-frontend/blob/e49b752801f61e33e214508d8424422d63cd81f9/src/features/Catalog/components/Main/ProductCard.tsx#L84-L86" class="text-red">84-86</a>, <a href="https://github.com/Licruit/Licruit-frontend/blob/e49b752801f61e33e214508d8424422d63cd81f9/src/features/Catalog/components/Main/ProductCard.tsx#L94" class="text-red">94</a>, <a href="https://github.com/Licruit/Licruit-frontend/blob/e49b752801f61e33e214508d8424422d63cd81f9/src/features/Catalog/components/Main/ProductCard.tsx#L99" class="text-red">99</a>, <a href="https://github.com/Licruit/Licruit-frontend/blob/e49b752801f61e33e214508d8424422d63cd81f9/src/features/Catalog/components/Main/ProductCard.tsx#L104" class="text-red">104</a>, <a href="https://github.com/Licruit/Licruit-frontend/blob/e49b752801f61e33e214508d8424422d63cd81f9/src/features/Catalog/components/Main/ProductCard.tsx#L109-L110" class="text-red">109-110</a>, <a href="https://github.com/Licruit/Licruit-frontend/blob/e49b752801f61e33e214508d8424422d63cd81f9/src/features/Catalog/components/Main/ProductCard.tsx#L114" class="text-red">114</a>, <a href="https://github.com/Licruit/Licruit-frontend/blob/e49b752801f61e33e214508d8424422d63cd81f9/src/features/Catalog/components/Main/ProductCard.tsx#L117" class="text-red">117</a></td> </tr> <tr> <td align="left"><a href="https://github.com/Licruit/Licruit-frontend/blob/e49b752801f61e33e214508d8424422d63cd81f9/src/features/Catalog/components/Main/ProductGrid.tsx">src/features/Catalog/components/Main/ProductGrid.tsx</a></td> <td align="right">0%</td> <td align="right">0%</td> <td align="right">0%</td> <td align="right">0%</td> <td align="left"><a href="https://github.com/Licruit/Licruit-frontend/blob/e49b752801f61e33e214508d8424422d63cd81f9/src/features/Catalog/components/Main/ProductGrid.tsx#L1-L5" class="text-red">1-5</a>, <a href="https://github.com/Licruit/Licruit-frontend/blob/e49b752801f61e33e214508d8424422d63cd81f9/src/features/Catalog/components/Main/ProductGrid.tsx#L7-L9" class="text-red">7-9</a>, <a href="https://github.com/Licruit/Licruit-frontend/blob/e49b752801f61e33e214508d8424422d63cd81f9/src/features/Catalog/components/Main/ProductGrid.tsx#L11-L15" class="text-red">11-15</a>, <a href="https://github.com/Licruit/Licruit-frontend/blob/e49b752801f61e33e214508d8424422d63cd81f9/src/features/Catalog/components/Main/ProductGrid.tsx#L17-L23" class="text-red">17-23</a>, <a href="https://github.com/Licruit/Licruit-frontend/blob/e49b752801f61e33e214508d8424422d63cd81f9/src/features/Catalog/components/Main/ProductGrid.tsx#L25-L33" class="text-red">25-33</a>, <a href="https://github.com/Licruit/Licruit-frontend/blob/e49b752801f61e33e214508d8424422d63cd81f9/src/features/Catalog/components/Main/ProductGrid.tsx#L35" class="text-red">35</a>, <a href="https://github.com/Licruit/Licruit-frontend/blob/e49b752801f61e33e214508d8424422d63cd81f9/src/features/Catalog/components/Main/ProductGrid.tsx#L37-L44" class="text-red">37-44</a>, <a href="https://github.com/Licruit/Licruit-frontend/blob/e49b752801f61e33e214508d8424422d63cd81f9/src/features/Catalog/components/Main/ProductGrid.tsx#L46" class="text-red">46</a>, <a href="https://github.com/Licruit/Licruit-frontend/blob/e49b752801f61e33e214508d8424422d63cd81f9/src/features/Catalog/components/Main/ProductGrid.tsx#L48" class="text-red">48</a>, <a href="https://github.com/Licruit/Licruit-frontend/blob/e49b752801f61e33e214508d8424422d63cd81f9/src/features/Catalog/components/Main/ProductGrid.tsx#L50" class="text-red">50</a>, <a href="https://github.com/Licruit/Licruit-frontend/blob/e49b752801f61e33e214508d8424422d63cd81f9/src/features/Catalog/components/Main/ProductGrid.tsx#L54" class="text-red">54</a></td> </tr> <tr> <td align="left"><a href="https://github.com/Licruit/Licruit-frontend/blob/e49b752801f61e33e214508d8424422d63cd81f9/src/pages/CatalogPage.tsx">src/pages/CatalogPage.tsx</a></td> <td align="right">0%</td> <td align="right">0%</td> <td align="right">0%</td> <td align="right">0%</td> <td align="left"><a href="https://github.com/Licruit/Licruit-frontend/blob/e49b752801f61e33e214508d8424422d63cd81f9/src/pages/CatalogPage.tsx#L1-L3" class="text-red">1-3</a>, <a href="https://github.com/Licruit/Licruit-frontend/blob/e49b752801f61e33e214508d8424422d63cd81f9/src/pages/CatalogPage.tsx#L5-L16" class="text-red">5-16</a>, <a href="https://github.com/Licruit/Licruit-frontend/blob/e49b752801f61e33e214508d8424422d63cd81f9/src/pages/CatalogPage.tsx#L18" class="text-red">18</a>, <a href="https://github.com/Licruit/Licruit-frontend/blob/e49b752801f61e33e214508d8424422d63cd81f9/src/pages/CatalogPage.tsx#L20" class="text-red">20</a>, <a href="https://github.com/Licruit/Licruit-frontend/blob/e49b752801f61e33e214508d8424422d63cd81f9/src/pages/CatalogPage.tsx#L22-L23" class="text-red">22-23</a>, <a href="https://github.com/Licruit/Licruit-frontend/blob/e49b752801f61e33e214508d8424422d63cd81f9/src/pages/CatalogPage.tsx#L34" class="text-red">34</a></td> </tr> </tbody> </table></details>
<em>Generated in workflow <a href=https://github.com/Licruit/Licruit-frontend/actions/runs/10303307265>#73</a></em>

File: src/features/Catalog/components/Header/CatalogHeader.tsx [hunk 1]
```diff
@@ -22,6 +21,15 @@ const Container = styled.div`
...
   border-bottom: 1px solid ${({ theme }) => theme.color.neutral[400]};
+  position: -webkit-sticky;
+  position: sticky;
+  top: 76px;
+  left: 0;
+  z-index: 99;
+
+  background: rgba(255, 255, 255, 0.6);
+  backdrop-filter: blur(10px);
+  -webkit-backdrop-filter: blur(10px);
```
Comment (JIMIN1020) [inline src/features/Catalog/components/Header/CatalogHeader.tsx:32]: 카탈로그 목록 페이지는 페이지 자체가 고정되어있고 그리드 쪽만 스크롤 되는거 아니었나요??!

**Review by SebellKo — APPROVED** [review]
주히님 엄청 고생하셨습니다 !

저도 빨리 끝내고 눕고싶네요 ㅎㅎ
남은 작업은 푹 주무시고 일어나셔서 하세요 ~

**Review by JIMIN1020 — APPROVED** [review]
고생 많으셨습니다~!

File: src/features/Catalog/components/Header/CatalogHeader.tsx (hunk 1 above)
Comment (juhee067) [reply src/features/Catalog/components/Header/CatalogHeader.tsx:32]: 그 부분 몰랐던 부분인데 UI가 그게 훨씬 낫겠네요 
그렇다면 맨 하단에 보여지는 footer 스크롤이 끝난후 왼쪽 필터와 헤더가 sticky로 설정이 돼서 내려와야하는지 고민이네요

File: src/features/Catalog/components/Header/Category.tsx [hunk 2]
```diff
@@ -28,6 +29,9 @@ function Category() {
           $style='outlined'
           $size='sm'
           onClick={() => {
```
Comment (ksy90101) [inline src/features/Catalog/components/Header/Category.tsx:31]: 여기 함수또한 함수로 추출하는게 훨씬 깔끔하거 같아요.
html에는 최대한 html의 코드만 있는게 나을거 같습니다.

File: src/features/Catalog/components/Main/ProductGrid.tsx [hunk 3]
```diff
@@ -54,6 +61,13 @@ const Container = styled.div`
...
-  justify-content: space-between;
   margin-bottom: 30px;
+
+  @media (min-width: 768px) {
+    justify-content: space-around;
+  }
+
+  @media (min-width: 1024px) {
+    justify-content: space-between;
+  }
```
Comment (ksy90101) [inline src/features/Catalog/components/Main/ProductGrid.tsx:72]: 이거 중복아닌가요?

하나만 있어도 될거 같은데요.

**Pull request merged**